cmd/check.go

```diff
@@ -328,7 +337,23 @@ func check() {
 			input[dg] = information[dg]
 		}
 
-		rc, err := packaging.EvalPackage(ctx, pkg, input)
+		// Determine which rules should be enabled
+		enabledRuleIDs := map[string]bool{}
+		ruleEnabledDefault := true
+		if len(enabledPackage.EnabledRules) != 0 {
+			ruleEnabledDefault = false
+		}
+		for _, ruleID := range manifest.RuleIDs() {
+			enabledRuleIDs[ruleID] = ruleEnabledDefault
+		}
+		for _, enabledRule := range enabledPackage.EnabledRules {
+			enabledRuleIDs[enabledRule] = true
+		}
+		for _, disabledRule := range enabledPackage.DisabledRules {
+			enabledRuleIDs[disabledRule] = false
+		}
+
+		rc, err := packaging.EvalPackage(ctx, pkg, input, enabledRuleIDs)
```

Comment:
I'd avoid dealing with the filter during the evaluation at all.

I'd rather filter the `ResultCollection` after the evaluation. That will keep the `packaging` package completely agnostic from this change, which is good for simplicity.

Something like this would be my preference:
```
type Filter struct {
  Package string
  EnabledRules []string
  DisabledRules []string
}

// FilterResults filters the results using a filter. If filter is nil, the same ResultCollection will be returned.
func FilterResults(rc *ResultCollection, filter *Filter) (*ResultCollection, error) {
}
```

It also makes sense to convert `FilterResults` into a function for the type `ResultCollection`.

Making the filter part of the `results` package is convenient so it can be reused as a separate package (ideally, `cmd` should be as simple as possible).

Reply:
Yeah that sounds sensible, I'll change that now.

Reply:
So I have moved the filtering to a separate function, but made . it just take two string slices rather than moving the struct because I plan for the struct to form part of the config struct once we start on #64.